### PR TITLE
WIP: Debug ResourceWarnings BufferedRandom

### PR DESCRIPTION
**Pull request by mroeschke**
xref #44844

### CODE REVIEW

**Comment by twoertwein** [comment]
might be good to have a test whether `lsof` finds ResourceWarnings (the psutil doesn't seem to find them/some of them).

```py
with pytest.raises(...):
    with check_bufferedrandom_resourcewarning():
        foo = open("foo")
foo.close()
```

Specifically enabling ResourceWarnings (usually disabled by default) seemed to work well for me in #39047: adding `always::ResourceWarning` to `filterwarnings` in setup.cfg.

File: pandas/conftest.py [hunk 1]
```diff
@@ -199,6 +200,20 @@ def add_imports(doctest_namespace):
     doctest_namespace["pd"] = pd
 
 
+pat = re.compile(r"(0?[1-9]|1[0-9]|2[0-5])u")
+
+
+@pytest.fixture(autouse=True)
```
Comment (jbrockmendel) [inline pandas/conftest.py:206]: maybe just enable for pandas.tests.io?

**Comment by mroeschke** [comment]
> might be good to have a test whether `lsof` finds ResourceWarnings (the psutil doesn't seem to find them/some of them).
> 
> ```python
> with pytest.raises(...):
>     with check_bufferedrandom_resourcewarning():
>         foo = open("foo")
> foo.close()
> ```
> 
> Specifically enabling ResourceWarnings (usually disabled by default) seemed to work well for me in #39047: adding `always::ResourceWarning` to `filterwarnings` in setup.cfg.

Yeah might do this in a follow up. Ideally just checking which tests open up new OS-level file handles. Apparently the `BufferedResourceWarning` opens it up at n=10

**Comment by mroeschke** [comment]
Closing in favor of https://github.com/pandas-dev/pandas/pull/45073

**Pull request closed without merging**